flask_rebar/swagger_generation/swagger_generator_v2.py

```diff
@@ -116,20 +121,20 @@ def generate(
         :param bool sort_keys: Use OrderedDicts sorted by keys instead of dicts
         :rtype: dict
         """
-        if registry.default_authenticators:
-            default_security = [
-                self.authenticator_converter.get_security_requirement(
-                    default_authenticator
-                )[0]
-                for default_authenticator in registry.default_authenticators
-                if default_authenticator is not None
-            ]
-        else:
-            default_security = []
-
-        security_definitions = self.authenticator_converter.get_security_schemes(
-            registry
-        )
+
+        security_definitions = {}
+        authenticators = get_unique_authenticators(registry)
+        for authenticator in authenticators:
+            security_definitions.update(
+                self.authenticator_converter.get_security_schemes(authenticator)
+            )
+
+        default_security = []
+        for authenticator in registry.default_authenticators:
+            default_security.extend(
+                self.authenticator_converter.get_security_requirements(authenticator)
+            )
```

Comment:
So if I am not mistaken we only support the OR case:
```
    security:    # A OR B
      - A
      - B
```
Since OpenAPI supports more than that, I would like to make a case that we also support the combination of auth. Though this is out of scope for this PR.

Reply:
My thought was that since Authenticators are classes you can add as much complexity there as you want. (I don't think we ever want to limit the functionality of an authenticator to a validating a single security definition.)

Given that you had Authenticator A and Authenticator B, and for one endpoint you wanted to require A && B, you could already create an Authenticator C which combines A and B.

With this change you can now also define the correct swagger security objects for that endpoint, which is what was not possible before.

So while I agree that it might be nice to have better support for `A AND B`, this should mean that we at least can properly support it; even if it's a bit dirty.

Reply:
(But also I'm kicking myself for not considering that when working on multiple Authenticator support since it probably would not have been too much extra work to support.)

Reply:
Agreed on that, maybe we should add documentation or examples on how to do that?